app/upgrades/v1-beta.4/bank_send.go

```diff
@@ -0,0 +1,237 @@
+	ctx.Logger().Info(fmt.Sprintf("bank send : total supply[%d]", before_total_supply.Amount))
+
+	// others
+	for index, value := range bank_send_list.Others {
+		fromAddr, _ := sdk.AccAddressFromBech32(value.FromAddress)
+		bankTarget := value.BankSendTarget
+		ctx.Logger().Info(fmt.Sprintf("bank send :%s", strconv.Itoa(index)))
+		normalToken := sdk.NewCoin(bankTarget.Denom, sdk.NewInt(bankTarget.Amount))
+		toAddr, _ := sdk.AccAddressFromBech32(bankTarget.ToAddress)
+		if err := bankkeeper.SendCoins(ctx, fromAddr, toAddr, sdk.NewCoins(normalToken)); err != nil {
+			panic(err)
+		}
+		total_allocate_coin.Add(normalToken)
+	}
+
+	// Check the amount of tokens sent
+	assumed_coin.Add(sdk.NewCoin(Denom, sdk.NewInt(TotalAmountTransferredValidator)))
+	assumed_coin.Add(sdk.NewCoin(Denom, sdk.NewInt(TotalDelegationAmountValidator)))
+	if !total_allocate_coin.IsEqual(assumed_coin) {
+		panic(fmt.Sprintf("error: assumed amount sent to the others validator does not match.: Actual[%d] Assumed[%d]",
+			total_allocate_coin.Amount,
+			assumed_coin.Amount))
+	}
+
+	fromAddr, err := sdk.AccAddressFromBech32(FromAddressValidator)
+	if err != nil {
+		panic(err)
+	}
+	// Validator
+	for index, value := range bank_send_list.Validator {
+		ctx.Logger().Info(fmt.Sprintf("bank send validator :%s[%s]", strconv.Itoa(index), value.ToAddress))
+		coin := tokenAllocation(ctx, authkeeper, bankkeeper, index, value, fromAddr)
+		total_allocate_coin.Add(coin)
+		normalToken := sdk.NewCoin(Denom, sdk.NewInt(FundAmountValidator))
+		toAddr, _ := sdk.AccAddressFromBech32(value.ToAddress)
+		if err := bankkeeper.SendCoins(ctx, fromAddr, toAddr, sdk.NewCoins(normalToken)); err != nil {
+			panic(err)
+		}
+		total_allocate_coin.Add(normalToken)
+	}
+
+	// Lend validatos
+	for index, value := range bank_send_list.LendValidator {
+		ctx.Logger().Info(fmt.Sprintf("bank send validator :%s[%s]", strconv.Itoa(index), value.ToAddress))
+		coin := tokenAllocation(ctx, authkeeper, bankkeeper, index, value, fromAddr)
+		total_allocate_coin.Add(coin)
+	}
+
+	// Check the amount of tokens sent
+	assumed_coin.Add(sdk.NewCoin(Denom, sdk.NewInt(TotalAmountValidator)))
+	if !total_allocate_coin.IsEqual(assumed_coin) {
+		panic(fmt.Sprintf("error: assumed amount sent to the validator does not match.: Actual[%d] Assumed[%d]",
+			total_allocate_coin.Amount,
+			assumed_coin.Amount))
+	}
+
+	fromAddr, err = sdk.AccAddressFromBech32(FromAddressEcocsytemDevelopment)
+	if err != nil {
+		panic(err)
+	}
+	// Ecocsytem Development(Community Program, Competition, Moderator)
+	for index, value := range bank_send_list.EcocsytemDevelopment {
+		ctx.Logger().Info(fmt.Sprintf("bank send :%s", strconv.Itoa(index)))
+		coin := tokenAllocation(ctx, authkeeper, bankkeeper, index, value, fromAddr)
+		total_allocate_coin.Add(coin)
+	}
+
+	// Check the amount of tokens sent
+	assumed_coin.Add(sdk.NewCoin(Denom, sdk.NewInt(TotalAmountEcocsytemDevelopment)))
+	if !total_allocate_coin.IsEqual(assumed_coin) {
```

Comment:
@taiki1frsh 
The checking part here is not working, so we are fixing it.
For total_allocate_coin and assumed_coin, change the method to check by adding the Amount by int64 instead of sdk.Coin.

Reply:
got it.

Reply:
@taiki1frsh 
TotalAmountEcocsytemDevelopment still contains the Vanya portion of the Amount, and the list should have been panic when checking because it omits the Vanya portion, but currently it is not working well because it is not an error.

TotalAmountEcocsytemDevelopment still includes Vanya's amount.
The bank send list omits the Vanya amount.

Reply:
changed a variable and committed

Reply:
@hikaruNagamine 
Does changing sdk.Coin to int make any difference here?

Reply:
@taiki1frsh 
You are right.
I went back and checked again and realized I had made an elementary mistake.
I have reverted back to using sdk.Coin.
The cause was not re-storing the return value into a variable after Add.

Reply:
got it.